### PR TITLE
updated checkstyle plugin

### LINKED ISSUE #192: Update checkstyle plugin
#### Problem description
We haven't upgraded the `checkstyle` plugin for quite some time. Now, we should use the latest version to include all of the new features added recently

#### Steps to reproduce the problem
Not an issue, code maintenance

#### Expected behavior
`quality.gradle` contains the configuration for `checkstyle` plugin, that should be upgraded. All of the introduced modules should be added to `checkstyle.xml`

#### Other information 
Things you tried, stack traces, related issues, suggestions on how to fix it...

### PR DESCRIPTION
**Pull request by iamvasuh**
Closes #192 

#### What has been done to verify that this works as intended?
checkstyle version has been updated 

#### Why is this the best possible solution? Were any other approaches considered?
this is the best solution
#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
code check mechanism is up-to-date
#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkCode` and confirmed all checks still pass OR confirm CircleCI build passes
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/opendatakit/share/blob/master/share_app/src/main/assets/open_source_licenses.html).

### CODE REVIEW

**Comment by iamvasuh** [comment]
i have been seeing this fix for the root module error
https://stackoverflow.com/questions/49645147/checkstyle-unable-to-create-root-module

**Comment by lakshyagupta21** [comment]
@iamvasuh  I saw that you closed this PR, is there any specific reason for closing this PR.

**Comment by iamvasuh** [comment]
> @iamvasuh I saw that you closed this PR, is there any specific reason for closing this PR.

This PR didn't pass all the checks and I tried to fix the check issue but it's not getting fixed.

File: skunkworks_crow/src/main/java/org/odk/share/activities/InstancesList.java [hunk 1]
```diff
@@ -134,8 +134,7 @@ public void toggle() {
             if (cursor.moveToFirst()) {
                 do {
                     selectedInstances.add(cursor.getLong(cursor.getColumnIndex(InstanceProviderAPI.InstanceColumns._ID)));
-                } while (cursor.moveToNext());
-            }
+                } while (cursor.moveToNext());}
```
Comment (rohap97) [inline skunkworks_crow/src/main/java/org/odk/share/activities/InstancesList.java]: Add a space after semi-colon - Line 137.

File: skunkworks_crow/src/main/java/org/odk/share/fragments/BlankFormsFragment.java [hunk 2]
```diff
@@ -175,8 +175,7 @@ public void toggle() {
             if (cursor.moveToFirst()) {
                 do {
                     selectedForms.add(cursor.getLong(cursor.getColumnIndex(FormsProviderAPI.FormsColumns._ID)));
-                } while (cursor.moveToNext());
-            }
+                } while (cursor.moveToNext());}
```
Comment (rohap97) [inline skunkworks_crow/src/main/java/org/odk/share/fragments/BlankFormsFragment.java]: Add a space after semi-colon - Line 178

File: skunkworks_crow/src/main/java/org/odk/share/fragments/FilledFormsFragment.java [hunk 3]
```diff
@@ -168,15 +168,13 @@ public void toggle() {
...
-            }
+                } while (cursor.moveToNext()); }
         } else {
             selectedInstances.clear();
         }
 
         instanceAdapter.notifyDataSetChanged();
-        toggleButtonLabel();
-    }
+        toggleButtonLabel();}
```
Comment (rohap97) [inline skunkworks_crow/src/main/java/org/odk/share/fragments/FilledFormsFragment.java]: Add a space after semi-colon. Line - 177

**Comment by rohap97** [comment]
After these changes, the build should pass.

**Comment by iamvasuh** [comment]
> Your build is failing due to there is no whitespace you can resolve it easily by select all the code and just reformat it using Android studio then the build will pass

i will just check it out. thank u

**Comment by iamvasuh** [comment]
> After these changes, the build should pass.

thank u so much

**Comment by iamvasuh** [comment]
@rohap97 @Chromicle i have tried both the ways , still having issues

**Comment by shobhitagarwal1612** [comment]
I think we need to a step back and look at what we want to achieve and what's the right way to do it.

So, the issue says that the plugin can be updated to the latest version by updating the version declared in quality.gradle. So that's step 1

Next, if you run the `checkstyle` gradle task after this, you will notice that the build is failing due to the addition of new checks. So, step 2 would be to either fix those issues (which I don't recommend doing in the same PR) or exclude those checks to make the build pass and fix them later in a separate issue.

Therefore, you need to do the following things:
 - Update the plugin 
 - Exclude the new code quality checks to make the build pass if it is failing

**Comment by iamvasuh** [comment]
> I think we need to a step back and look at what we want to achieve and what's the right way to do it.
> 
> So, the issue says that the plugin can be updated to the latest version by updating the version declared in quality.gradle. So that's step 1
> 
> Next, if you run the `checkstyle` gradle task after this, you will notice that the build is failing due to the addition of new checks. So, step 2 would be to either fix those issues (which I don't recommend doing in the same PR) or exclude those checks to make the build pass and fix them later in a separate issue.
> 
> Therefore, you need to do the following things:
> 
> * Update the plugin
> * Exclude the new code quality checks to make the build pass if it is failing

exactly @shobhitagarwal1612 it just came to my mind after seeing your comment on #193

**Comment by iamvasuh** [comment]
@shobhitagarwal1612  thank u so much

File: config/quality.gradle [hunk 4]
```diff
@@ -21,8 +21,6 @@ def reportsDir = "${project.buildDir}/reports"
 
 apply plugin: 'checkstyle'
 
-checkstyle.toolVersion = '7.6.1'
```
Comment (shobhitagarwal1612) [inline config/quality.gradle]: Removing this line updates the checkstyle version to the latest?
How did you verify that it indeed used the latest version?

Comment (iamvasuh) [reply config/quality.gradle]: as per @huangyz0918  suggestion, i had it removed and let the android studio sync the gradle

Comment (iamvasuh) [reply config/quality.gradle]: in my #200 , i had it updated to 8.18

Comment (huangyz0918) [reply config/quality.gradle]: I have to clarify, @iamvasuh I mean, the default version of checkstyle depends on your gradle version, it's not `the` latest among all the checkstyle versions. it will update automatically when your gradle updates.

Comment (huangyz0918) [reply config/quality.gradle]: You can refer to my [comment](https://github.com/opendatakit/skunkworks-crow/pull/200#issuecomment-474251455),

>for example, from Gradle 3.3 onwards, the default checkstyle version is 6.19; before, it was 5.9. Only Gradle versions prior to 2.4 used even older versions of checkstyle.

> So, as we mentioned above, we can use the default checkstyle version, it will auto update as our Gradle version updates, or, we can just update the checkstyle version to 8.18 like your commit.

Comment (iamvasuh) [reply config/quality.gradle]: i checked my gradle version, its 4.4 right now , so after making changes to the checkstyle plugin version(i.e. removing the version no. , excluding the whitespace modules from the checkstyle.xml file and letting the studio do its gradle sync ) it's verified that the project is using the latest version of checktyle plugin. @shobhitagarwal1612 @huangyz0918 
Screen Shot 2019-03-22 at 10 35 36 AM

File: config/checkstyle.xml [hunk 5]
```diff
@@ -67,17 +67,6 @@
...
-            <property name="id" value="RightCurlySame" />
-            <property name="tokens"
-                value="LITERAL_TRY, LITERAL_CATCH, LITERAL_FINALLY, LITERAL_IF, LITERAL_ELSE, LITERAL_DO" />
-        </module>
-        <module name="RightCurly">
-            <property name="id" value="RightCurlyAlone" />
-            <property name="option" value="alone" />
-            <property name="tokens"
-                value="CLASS_DEF, METHOD_DEF, CTOR_DEF, LITERAL_FOR, LITERAL_WHILE, STATIC_INIT, INSTANCE_INIT" />
-        </module>
```
Comment (shobhitagarwal1612) [inline config/checkstyle.xml]: Why are these checks removed?

File: skunkworks_crow/src/main/java/org/odk/share/activities/InstancesList.java [hunk 6]
```diff
@@ -32,21 +32,21 @@
...
-    @BindView(R.id.recyclerview) RecyclerView recyclerView;
-    @BindView(R.id.toolbar) Toolbar toolbar;
-    @BindView(R.id.send_button) Button sendButton;
-    @BindView(R.id.toggle_button) Button toggleButton;
-
-    private static final String INSTANCE_LIST_ACTIVITY_SORTING_ORDER = "instanceListActivitySortingOrder";
-
     public static final String INSTANCE_IDS = "instance_ids";
-
+    private static final String INSTANCE_LIST_ACTIVITY_SORTING_ORDER = "instanceListActivitySortingOrder";
```
Comment (shobhitagarwal1612) [inline skunkworks_crow/src/main/java/org/odk/share/activities/InstancesList.java]: Why have you modified this file? There doesn't seem to be any fix related to code quality. Please revert the changes to this file if not required

File: skunkworks_crow/src/main/java/org/odk/share/fragments/BlankFormsFragment.java [hunk 7]
```diff
@@ -175,7 +175,7 @@ public void toggle() {
             if (cursor.moveToFirst()) {
                 do {
                     selectedForms.add(cursor.getLong(cursor.getColumnIndex(FormsProviderAPI.FormsColumns._ID)));
-                } while (cursor.moveToNext());
+                } while (cursor.moveToNext()) ;
```
Comment (shobhitagarwal1612) [inline skunkworks_crow/src/main/java/org/odk/share/fragments/BlankFormsFragment.java]: Please revert this change...

**Comment by lakshyagupta21** [comment]
@iamvasuh If you're still working on this then please resolve the conflicts as well as fix the review comments.

**Comment by lakshyagupta21** [comment]
@iamvasuh if you still wish to continue on this please reopen it, for now, I'm closing this PR so that others can pick it up.
